kolena/_experimental/data_ingestion/_data_format/coco.py

```diff
@@ -0,0 +1,255 @@
+        self.raw_data = None
+        self.raw_inferences = None
+        for data in loaded_data:
+            if isinstance(data, list):
+                self.raw_inferences = data
+            if isinstance(data, dict):
+                self.raw_data = data
+        assert self.raw_data is not None
+        assert self.raw_inferences is not None
+
+    def _process_data(self) -> None:
+        self.image_map: Dict[int, Any] = self._get_image_map(self.raw_data)
+        self.label_map: Dict[int, str] = self._get_label_map(self.raw_data)
+        self.inf_map = self._get_inf_map(self.raw_inferences, self.image_map, self.label_map)
+
+    def _get_image_map(self, raw_data):
+        return {int(record["id"]): record for record in raw_data["images"]}
+
+    def _get_label_map(self, raw_data):
+        return {int(record["id"]): record["name"] for record in raw_data["categories"]}
+
+    def _get_inf_map(self, raw_inferences, image_map, label_map):
+        ids = set(image_map.keys())
+        image_to_boxes: Dict[int, List[ScoredLabeledBoundingBox]] = defaultdict(lambda: [])
+        for inference in raw_inferences:
+            image_id = inference["image_id"]
+            label = label_map[inference["category_id"]]
+
+            # check that the box is in a valid image
+            if image_id in ids:
+                bbox = inference["bbox"]
+                top_left = (bbox[0], bbox[1])
+                bottom_right = (bbox[0] + bbox[2], bbox[1] + bbox[3])
+                score = inference["score"]
+                bounding_box = ScoredLabeledBoundingBox(top_left, bottom_right, label, score)
+                image_to_boxes[image_id].append(bounding_box)
+        return image_to_boxes
+
+    def _setup_evaluator(self) -> ObjectDetectionEvaluator:
+        return ObjectDetectionEvaluator(
+            configurations=[
+                ThresholdConfiguration(
+                    threshold_strategy=0.3,
+                    iou_threshold=0.3,
+                    with_class_level_metrics=True,
+                    min_confidence_score=0.2,
+                ),
+                ThresholdConfiguration(
+                    threshold_strategy=F1_OPTIMAL,
+                    iou_threshold=0.5,
+                    with_class_level_metrics=True,
+                    min_confidence_score=0.0,
+                ),
+            ],
+        )
+
+    def _save_data(self) -> None:
+        model_name = self.config.model_name
+        test_suite_name = self.config.test_suite_name
+        reset = self.config.reset
+        # create a model
+        model = Model(model_name, infer=self._get_infer(self.inf_map))
+
+        # customizable configurations for the evaluator
+        evaluator = self._setup_evaluator()
+
+        # runs the evaluation
+        test_suite = TestSuite(test_suite_name)
+        test(model, test_suite, evaluator, reset=reset)
+
+    def _get_image_id(self, locator: str) -> int:
```

Comment:
This seems very fragile -- I can't imagine expecting customer data (set up as COCO JSON data) to adhere to this type of structure.